src/api/spec/controllers/staging/staging_projects_controller_spec.rb

```diff
@@ -138,6 +139,14 @@
                                                                                                            staging_project_copy_name,
                                                                                                            user.id)
     end
+
+    it 'does not allow unauthoried access' do
```

Comment:
Please do a policy test instead. You then can make sure the policy is applied in this controller with [`after_action :verify_authorized`](https://github.com/varvet/pundit#ensuring-policies-and-scopes-are-used) and if you really must, test here that this [`after_action`](http://matchers.shoulda.io/docs/v3.1.1/Shoulda/Matchers/ActionController.html#use_after_action-instance_method) is called.

Reply:
I find this really questionable. Because the policy did its job and the after_action did its job. Still the developer didn't realize *which* policies to apply. And there a test case can help the review process.

Reply:
> Still the developer didn't realize which policies to apply.

Can't save yourself from stupidity (you end up with what we have right now). This should have been catched during code-review, it has not been. The solution is to (help you) to review more carefully.

Reply:
As long as there are two authorization calls in the controller `verify_authorized` won't help much. The solution seems to be to have a staging project policy.

Reply:
Or we create a project policy for the `copy` action. :thinking:

Reply:
> As long as there are two authorization calls in the controller verify_authorized won't help much.

Like I said, you can't technically guard yourself from everything.

> The solution seems to be to have a staging project policy.

True, the check in the second `authorize` should move in there :+1:

Reply:
> Like I said, you can't technically guard yourself from everything.

But in this special case (having two authorize calls in one action) it seems to make sense to have 

> True, the check in the second authorize should move in there :+1:

I would say that either we move the whole authorization code there and only authorize it once or we live with the duplicated call for now (and use the same project policy).

I would go with the latter for now. Given that staging projects are currently no model of their own but a module added to a project.